### ISSUE TITLE
[BUG] (premature cut off of the heaters)

### ISSUE THREAD

**Issue by dsvi** [MarlinFirmware/Marlin#19317]
### Bug Description

It seems like motion commands get into buffer, but heating control commands are executed immediately. 
If you set `BLOCK_BUFFER_SIZE` big enough (128) or you have simple enough moves at the end of the job, heating turns off way before the job is finished. 

### My Configurations

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/5194609/Marlin.zip)

### Steps to Reproduce

Basically either set BLOCK_BUFFER_SIZE to something big, or print something simple enough (a cube). Better yet do both. See how heaters are turned off way before the print job ends.
In extreme cases this causes failed prints at the very end. Since plastic is not extruded anymore through cold nozzle.

**Comment by thisiskeithb** [comment]
Use [`M400`](https://marlinfw.org/docs/gcode/M400.html) before the disable heaters codes in your end script.
<hr>
This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- MarlinFW.org [Marlin Documentation](http://marlinfw.org)
- RepRap.org [Marlin Forum](http://forums.reprap.org/list.php?415)
- [Tom's 3D Forums](https://forum.toms3d.org/)
- Facebook Group ["Marlin Firmware"](https://www.facebook.com/groups/1049718498464482/)
- Facebook Group ["Marlin Firmware for 3D Printers"](https://www.facebook.com/groups/3Dtechtalk/)
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- Marlin Discord server. Join link: https://discord.gg/n5NJ59y

After seeking help from the community, if the consensus points to a bug in Marlin, then you should post a [bug report](https://github.com/MarlinFirmware/Marlin/issues/new/choose).
